service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -280,6 +312,10 @@ func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) getTargetClusterAndI
 		// fallback to selected APIs if targetCluster is not empty and not the same as currentActiveCluster
 	}
 
+	if requestedConsistencyLevel == types.QueryConsistencyLevelStrong {
```

Comment:
you can remove `QueryWorkflowStrongConsistency` from the `selectedAPIs*` maps at the top since it's handled by this condition now